src/upgini/autofe/feature.py

```diff
@@ -139,27 +122,27 @@ def get_display_name(self, cache: bool = True, shorten: bool = False, **kwargs):
             self.cached_display_name = display_name
         return display_name
 
-    def set_display_index(self, index):
+    def set_display_index(self, index) -> "Feature":
         self.display_index = index
         self.cached_display_name = None
         return self
 
-    def infer_type(self, data):
+    def infer_type(self, data: pd.DataFrame) -> Union[str, DtypeObj]:
         if self.op.output_type:
             return self.op.output_type
         else:
             # either a symmetrical operator or group by
             return self.children[0].infer_type(data)
 
-    def calculate(self, data, is_root=False):
+    def calculate(self, data: pd.DataFrame, is_root=False) -> Union[pd.Series, pd.DataFrame]:
         if isinstance(self.op, PandasOperand) and self.op.is_vector:
-            ds = [child.calculate(data) for child in self.children]
-            new_data = self.op.calculate(data=ds)
-
-        elif isinstance(self.op, PandasOperand):
-            d1 = self.children[0].calculate(data)
-            d2 = None if len(self.children) < 2 else self.children[1].calculate(data)
-            new_data = self.op.calculate(data=d1, left=d1, right=d2)
+            if self.op.is_vector:
+                ds = [child.calculate(data) for child in self.children]
+                new_data = self.op.calculate(data=ds)
+            else:
```

Comment:
If you're doing this, you should also remove self.op.is_vector check on line 138. Otherwise we lose a branch for PandasOperators which are not vector

Reply:
Fixed. I've just forgot to remove this check from outer if